crypto/src/test_utils/zero_crypto.rs

```diff
@@ -0,0 +1,400 @@
+
+pub type ZeroPublicKeyBytes = PublicKeyBytes<ZeroPublicKey, { PUBLIC_KEY_LENGTH }>;
+
+#[derive(Default, Debug)]
+pub struct ZeroPrivateKey(pub [u8; PRIVATE_KEY_LENGTH]);
+
+// There is a strong requirement for this specific impl. in Fab benchmarks
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(tag = "type")] // necessary so as not to deser under a != type
+pub struct ZeroKeyPair {
+    name: ZeroPublicKey,
+    secret: ZeroPrivateKey,
+}
+
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct ZeroSignature {}
+
+#[serde_as]
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct ZeroAggregateSignature {}
+
+///
+/// Implement SigningKey
+///
+
+impl AsRef<[u8]> for ZeroPublicKey {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl ToFromBytes for ZeroPublicKey {
+    fn from_bytes(bytes: &[u8]) -> Result<Self, signature::Error> {
+        let bytes_fixed: [u8; PUBLIC_KEY_LENGTH] =
+            bytes.try_into().map_err(|_| signature::Error::new())?;
+        Ok(Self(bytes_fixed))
+    }
+}
+
+impl std::hash::Hash for ZeroPublicKey {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        self.as_ref().hash(state);
+    }
+}
+
+impl PartialEq for ZeroPublicKey {
+    fn eq(&self, other: &Self) -> bool {
+        self.0 == other.0
+    }
+}
+
+impl Eq for ZeroPublicKey {}
+
+impl PartialOrd for ZeroPublicKey {
+    fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
+        self.as_ref().partial_cmp(other.as_ref())
+    }
+}
+impl Ord for ZeroPublicKey {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.as_ref().cmp(other.as_ref())
+    }
+}
+
+impl Display for ZeroPublicKey {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        write!(f, "{}", Base64::encode_string(self.as_ref()))
+    }
+}
+
+// There is a strong requirement for this specific impl. in Fab benchmarks
+impl Serialize for ZeroPublicKey {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        serializer.serialize_str(&self.encode_base64())
+    }
+}
+
+// There is a strong requirement for this specific impl. in Fab benchmarks
+impl<'de> Deserialize<'de> for ZeroPublicKey {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        let s = <String as serde::Deserialize>::deserialize(deserializer)?;
+        let value = Self::decode_base64(&s).map_err(|e| de::Error::custom(e.to_string()))?;
+        Ok(value)
+    }
+}
+
+impl Verifier<ZeroSignature> for ZeroPublicKey {
+    fn verify(&self, _msg: &[u8], _signature: &ZeroSignature) -> Result<(), signature::Error> {
+        Ok(())
+    }
+}
+
+impl<'a> From<&'a ZeroPrivateKey> for ZeroPublicKey {
+    fn from(secret: &'a ZeroPrivateKey) -> Self {
+        let mut hasher = Sha3_256::new();
+        hasher.update(secret.0);
```

Comment:
I like the approach of using the hash of priv key to derive the pub. Although we could in theory just copy the value to the pubKey, this function is not affecting our performance, so I agree with your logic.